### PR TITLE
create index on start up

### PR DESCRIPTION
**Pull request by Tianhao-Gu**

### CODE REVIEW

**Comment by Tianhao-Gu** [comment]
@MrCreosote @kkellerlbl Do you mind taking a look at this PR?

**Comment by MrCreosote** [comment]
Hey, while you're already working on hs2 what do you think about taking care of https://github.com/kbase/handle_service2/issues/93 as well? Should be an easy fix

File: test/MongoUtil_test.py [hunk 1]
```diff
@@ -62,6 +62,10 @@ def test_init_ok(self):
         self.assertEqual(handle_collection.name, 'handle')
         self.assertEqual(handle_collection.count_documents({}), 10)
 
+        handle_col_idx = handle_collection.index_information().keys()
+        expected_idx = ['_id_', 'hid_1']
+        self.assertCountEqual(handle_col_idx, expected_idx)
```
Comment (MrCreosote) [inline test/MongoUtil_test.py:67]: Maybe check it's unique as well

**Review by MrCreosote — CHANGES_REQUESTED** [review]
Looks good, just one small test addition

**Comment by MrCreosote** [comment]
I forgot one thing - can you add something to the release notes that tells the user they need to add a unique index on hid in the background prior to upgrade or the service will hang on startup until the index is built

File: test/MongoUtil_test.py (hunk 1 above)
Comment (Tianhao-Gu) [reply test/MongoUtil_test.py:68]: 👍

**Comment by Tianhao-Gu** [comment]
@MrCreosote Thanks. Do you mind taking another look?

File: test/MongoUtil_test.py [hunk 2]
```diff
@@ -62,9 +62,11 @@ def test_init_ok(self):
         self.assertEqual(handle_collection.name, 'handle')
         self.assertEqual(handle_collection.count_documents({}), 10)
 
-        handle_col_idx = handle_collection.index_information().keys()
+        handle_col_info = handle_collection.index_information()
+        handle_col_idx = handle_col_info.keys()
         expected_idx = ['_id_', 'hid_1']
         self.assertCountEqual(handle_col_idx, expected_idx)
```
Comment (MrCreosote) [inline test/MongoUtil_test.py:68]: Sorry, I missed this before - why assertCountEqual vs assertEqual?

**Review by MrCreosote — CHANGES_REQUESTED** [review]

File: test/MongoUtil_test.py (hunk 2 above)
Comment (Tianhao-Gu) [reply test/MongoUtil_test.py:68]: the `_id` is automatically indexed. Do you want me to change it to something like assertIn('hid_1', handle_col_idx) to only check `hid` is indexed?

Comment (Tianhao-Gu) [reply test/MongoUtil_test.py]: oh I think I see what you mean now. `handle_col_idx` was a dictkey. I was just lazy to cast it to a list. I can change it to use assertEqual.

Comment (Tianhao-Gu) [reply test/MongoUtil_test.py]: 👍

**Review by MrCreosote — APPROVED** [review]
LGTM

**Pull request merged**